js/actions/user.js

```diff
@@ -21,14 +16,34 @@ const RECEIVE_USER_PROFILE_ACTION = 'RECEIVE_USER_PROFILE_ACTION'
 const UPDATE_USER_PROFILE_REQUEST_ACTION = 'UPDATE_USER_PROFILE_REQUEST_ACTION'
 const UPDATE_USER_PROFILE_ERROR_ACTION = 'UPDATE_USER_PROFILE_ERROR_ACTION'
 
+// Creators
+function receiveUserProfile(profile: ApiUserProfile): Action {
```

Comment:
you should declare specific types for each action in `types.js` and here use the specific type for this action, e.g:

```js

// types.js

type ReceiveUserProfileAction = {
      type: 'RECEIVE_USER_PROFILE_ACTION',
      profile: ApiUserProfile,
      receivedAt: number
    };

type Action = ....
  | ReceiveUserProfileAction
  | ...;

// user.js

function receiveUserProfile(profile: ApiUserProfile): ReceiveUserProfileAction {
// ...
}

```

Reply:
I am already using this style on new actions i have created in the /js/store directory, the only difference is that types are not all in the same file (types.js) but each action file (like profile.js) exports own related actiontypes. Do you prefer a single types.js file with all the actiontypes?. I'll apply this also for the old login/user actions.

Reply:
yes it's preferable as it's a single place where one could see all the actions triggered by the application

Reply:
I have create a story to export all the action types to a single file https://www.pivotaltracker.com/story/show/156155528.